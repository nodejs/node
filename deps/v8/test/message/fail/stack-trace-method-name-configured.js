// Copyright 2022 the V8 project authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

// Check that the user configured method name ('SomeClass.someMethod') is used
// as-is in error.stack output without prepending the inferred type name ('A').
const A = function() {};
A.prototype.a = function() {
  throw new Error;
};
Object.defineProperty(A.prototype.a, 'name', {value: 'SomeClass.someMethod'});
(new A).a();
