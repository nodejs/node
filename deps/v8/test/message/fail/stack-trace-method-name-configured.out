# Copyright 2022 the V8 project authors. All rights reserved.
# Use of this source code is governed by a BSD-style license that can be
# found in the LICENSE file.

*%(basename)s:9: Error
  throw new Error;
  ^
Error
    at SomeClass.someMethod [as a] (*%(basename)s:9:9)
    at *%(basename)s:12:9

